### ISSUE TITLE
Some images processed by the crate end up corrupted

### ISSUE THREAD

**Issue by mwap** [image-rs/image#1359]
This happens when I read, decode, encode and save some of the images from a specific source: https://ant-photo.eu/galeria/zdjecia-mrowek

About 10% of the images get corrupted in the process and every time I run the program it's the same images that do. I don't see anything special about the input images that get corrupted. All corrupted output images have the same glitchy pattern and are about 7 times bigger in bytes.

## Expected

Output images are not corrupted.

## Actual behaviour

Output images are low quality and are visually corrupted (see below).

## Reproduction steps

1. Download this image: https://ant-photo.eu/wp-content/uploads/2020/07/Cataglyphis-iberica-worker2.jpg
2. Save it as `in.jpg`
3. Run the following code:
```rust
let mut image_data = image::io::Reader::open("in.jpg")?.decode()?;
image_data.save("out.jpg")?;
```
4. `out.jpg` looks like this: https://i.ibb.co/j8rdwNc/sand.jpg

**Comment by HeroicKatora** [comment]
The issue is also present when writing the data as `png` or other formats. It seems to be a problem in the decoder.

**Comment by HeroicKatora** [comment]
This should have been fixed in `jpeg-decoder` `v0.1.22`, it was one of the local test images.
